htlcswitch/link_test.go

```diff
@@ -6894,3 +6895,216 @@ func TestChannelLinkShortFailureRelay(t *testing.T) {
 	default:
 	}
 }
+
+func TestLinkDrainApiDirectionIsolation(t *testing.T) {
```

Comment:
FWIW, we usually only add tests in commits where they pass, can make CI happier with cherry-picks n such.

Reply:
No TDD?